lib/routes/token.js

```diff
@@ -53,6 +54,19 @@ const PAYLOAD_SCHEMA = Joi.object({
     }),
 
   client_secret: validators.clientSecret
+    //.required()
+    .when('grant_type', {
+      is: GRANT_JWT,
+      then: Joi.forbidden(),
+      //otherwise: Joi.required()
+    }),
+    // .when('code_verifier', {
+    //   is: Joi.string(),
+    //   then: Joi.forbidden(),
+    //   //otherwise: Joi.required()
+    // }),
```

Comment:
@seanmonstar any ideas here how to make `clientSecret` required in ALL cases except when `GRANT_JWT` or `code_verifier` is SET? The commented code above still wants `client_secret`